- change dbName

Sheydss opened the pull request:
- added restaurant class
- fix registrySubscription
- added verification of restaurant owner
- fix get by name or by restaurantid

Commit:
- change dbName
- added restaurant class
- fix registrySubscription
- added verification of restaurant owner
- fix get by name or by restaurantid

Asarck commented on src/views/menuView.js:
```diff
@@ -19,6 +20,11 @@ const errors = {
     })(),
+    Unauthorized: (() => {
+        const err = Error("Access denied");
+        err.statusCode = 401;
```
err.statusCode = 403 ?

Asarck commented on src/views/menuView.js:
```diff
@@ -138,18 +146,25 @@ module.exports = {
+        const restaurant = await Restaurant.findById(article.restaurantId);
+        if(req.query.roleLabel != 'admin' && req.query.userId != restaurant.restaurantOwnerId) return errors.Unauthorized;
+
         if(menu) {
```
if (!menu) -> logique erreur
pas besoin de else ensuite

Asarck commented on src/views/menuView.js:
```diff
@@ -188,14 +203,19 @@ module.exports = {
-        });
-
-        if(menu) return `productIdList in menu ${menuId} updated`;
+        if(menu) {
```
if (!menu) -> logique erreur
pas besoin de else ensuite

Asarck commented:
La logique me semble bonne de mon côté, juste de le refacto de code car je suis un hater d'indentation et à débattre pour le 401/403 en code d'erreur

Commit:
- Factorization based on comments
- changed code 401 to 403 :
401 the request lacks valid authentification credentials
403 server understand the request but refuses to fulfill it

Sheydss commented:
Corrigé les problèmes de factorisation et changé 401 en 403 car :
401 the request lacks valid authentification credentials
403 server understand the request but refuses to fulfill it
